sidebar/src/components/deploy/script/DeployUsingScript.logic.ts

```diff
@@ -21,21 +21,22 @@ export const useDeployUsingScript = (vscode: VSCode, scripts: Scripts, environme
       switch (event.data.type) {
         case MessageType.DEPLOY_SCRIPT_RESPONSE: {
           setResponse(event.data.response);
+          setIsPending(false);
           break;
         }
       }
     };
     window.addEventListener('message', listener);
     return () => window.removeEventListener('message', listener);
-  }, []);
+  }, [setIsPending]);
 
   useEffect(() => {
     form.setValue('script', scripts[0]?.id || '');
-  }, [scripts]);
+  }, [scripts, form]);
```

Comment:
Avoid using form in dependencies, otherwise the script will be set every time form change and we don't want that